### ISSUE TITLE
Project owner?

### ISSUE THREAD

**Issue by spaceone** [mamund/hal-forms#29]
Hello, I have questions about this project:

I found https://rwcbook.github.io/hal-forms/ which links to https://github.com/RWCBook/hal-forms.
And I see that this is the root github repository https://github.com/mamund/hal-forms/.

Who is the owner and founder? Why is that website not linking to this repository. Is it a fork with changes?

And even more interesting, who are the project founder? Are they involved in the IETF? Is there any chance that this specification becomes recognized by the IETF and maybe become a standard or at least a draft. Maybe it inspires HAL so that it goes directly into HAL.
Otherwise this is a nice project but unfortionately also a proprietrary data format.

**Comment by mamund** [comment]
Florian:

sorry for the confusion. here's the story

1) i am the owner/founder of HAL-FORMS
2) the *book* links point to a *fork* of the HAL-FORMS spec that is stable
[1] (I didn't want the spec ref'd in the book to change after publication.
3) the "home" repo for the spec is currently hosted in my own github org
[2]. this is the one that should be handling updates going foward.
4) eventually, i'd like to see HAL-FORMS as a registered media type (w/
IANA) in fact, the prs.* media identifier SHOULD be reg'd already. that's
just something i didn't get around to doing.
5) i'd like to see HAL-FORMS spec documented in a full IETF RFC. i wanted
to make sure the current HAL-FORMS has been vetted and has a stable set of
"version 1" features before starting the IEF work, tho. we might be there
or close by now.

i know the issues list is a bit old at the moment and needs some love. the
work of registering the prs* facet of the media type needs to be done now
and, once the spec is stable an RFC should be started w/ the IETF. I'm all
for moving ahead and would be happy to share the load on some of this work
if some people are willing to be co-sponsors of the spec.

hopefully, this clears things up a bit.

cheers.




[1] https://github.com/RWCBook/hal-forms
[2] https://github.com/mamund/hal-forms/.

Mike Amundsen

*APIs, Microservices, and Digital Transformation*
7310 Turfway Rd. Suite 550, Florence, KY, 41042


*+1.859.372.6517*
https://zoom.us/j/8593726715
http://linkedin.com/in/mamund
http://g.mamund.com/meetme
http://training.amundsen.com
http://twitter.com/mamund

**Comment by spaceone** [comment]
Thank you very much!
I will consider implementing HAL-FORMS and will soon give feedback by creating new issues or attaching to existing ones.

**Comment by mamund** [comment]
@spaceone 

thanks for the interest and please keep me posted on your experience.

cheers.

**Comment by spaceone** [comment]
@mamund 
FYI: I ialready invented my own format, before I heard of hal-forms.
If interested, you can see one example here:
[add.json](https://github.com/mamund/hal-forms/files/3536040/add.txt)

It basically uses _forms and describes all fields in there. From this data structure I am rendering a complete HTML website, so the API can be used both in a browser as well as via JSON consuming machines.
And for the GUI rendering I am defining _layout, which specifies the order and arrangement of properties in the field. This is not part of the json file currently.

**Comment by spaceone** [comment]
@mamund Maybe it's the wrong approach to invent more and more media types. It would be best to use the existing ones. Wouldn't it be possible to integrate hal-forms into `application/hal+json` and only specify some new link-relations, which are `_embedded` and have a certain format?

**Comment by mamund** [comment]
i'd be happy to see HAL adopt a FORMS approach. was clued in my Mike Kelly that he was not interested in that, so i went ahead and created this extension.

other's have press ahead anyway, i think.

**Comment by vpavic** [comment]
Hey @mamund, sorry to comment on a long closed issue, but do you think some of the info from [this comment](https://github.com/mamund/hal-forms/issues/29#issuecomment-524392801) of yours could be added to the project's readme?

IMO it would clear a lot of confusion between this repo and the [RWCBook/hal-forms](https://github.com/RWCBook/hal-forms) repo. And that's quite helpful for potential contributors.

Having contributed to this repo just yesterday, it was not so trivial to figure out where to direct my contribution, especially having in mind the entry point was the published version on https://rwcbook.github.io/hal-forms/.

**Comment by mamund** [comment]
may regrets for the confusion and for the late reply. I've updated the
RWCBOOK repo to remind people to post to the MAMUND repo instead.

thanks for the assist and please feel free to submit issues, PRs, etc. any
time.

@mamund
